core/src/main/java/hudson/model/View.java

```diff
@@ -1045,6 +1045,7 @@ public FormValidation doCheckJobName(@QueryParameter String value) {
      */
     @Restricted(DoNotUse.class)
     public Categories doItemCategories(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        getOwner().checkPermission(Item.CREATE);
```

Comment:
@daniel-beck 

> Because of folders.

Right, I always forget it

> Also, is there any reason to retrieve the data unless you need it for Create Item dialog?

I don't know, an API can be used on very different scenarios.